### DISCUSSION TITLE
Compiling a .js file and serving it later

### DISCUSSION THREAD

**Discussion by saleiv** [FranckFreiburger/vue3-sfc-loader#204] [Q&A]
Hey there FranckFreiburger,

First off the bat, I wanted to thank you for the amazing tool. It's great that you managed to connect so many different things into one functional repository.

We wanted to develop sites using primarily this and avoid the build step entirely. However, Google Page Speed insights has a bad score on Largest Contentful Paint due to processing occuring in the library.

Tried running the withCache example as a prototype, but it shoots an error on execution.

```
Uncaught (in promise) TypeError: null has no properties
    e vue3-sfc-loader.js:113
    e vue3-sfc-loader.js:113
    async*e</he/r[i]< vue3-sfc-loader.js:113
    he vue3-sfc-loader.js:113
    we vue3-sfc-loader.js:113
```

Is there a way to serialize the compiled modules somehow? I see that loadModule returns a promise with scopeId, render and setup functions - is there a way to serialize this and deserialize it later?

In essence, vue-sfc-compiler can compile a .js file that can be imported into the browser later on. Can a similar .js file be compiled somehow with your loader and then cached?

Any other approach or insight is very much appreciated!

Best regards,
Aleksandar

**Comment by FranckFreiburger** [comment]
Hi Saleiv

vue3-scf-loder cannot (an will not) pre-compile assets without they are used.
However, since v0.9.2+ vue3-scf-loder can run server-side to compile and cache assets.
see: https://github.com/FranckFreiburger/vue3-sfc-loader/discussions/178
